### PR TITLE
Feature/adding api service

### PR DESCRIPTION
**Pull request by HalynaHalkina**

### CODE REVIEW

File: app/src/main/java/com/uptech/example/vidme/api/dto/Auth.java [hunk 1]
```diff
@@ -2,15 +2,18 @@
...
  * Created by oleh on 31.03.16
  */
+@Getter
 public class Auth {
     private String token;
     private Date expieres;
     private int userId;
 
-    public String getToken() {
+    /*public String getToken() {
```
Comment (Dima564) [inline app/src/main/java/com/uptech/example/vidme/api/dto/Auth.java]: Remove

File: app/src/main/java/com/uptech/example/vidme/ui/front/FollowingTabFragment.java [hunk 2]
```diff
@@ -27,14 +24,12 @@ protected void loadVideos(int offset) {
...
-                .observeOn(AndroidSchedulers.mainThread())
-                .subscribe(videoListResponse -> {
-                    isLoading = false;
-                    adapter.addAll(videoListResponse.getVideos());
-                }, error -> {
-                    isLoading = false;
-                    makeToast(R.string.front_loading_list_error);
-                });
+        responseObservable.subscribe(videoListResponse -> {
+            isLoading = false;
```
Comment (Dima564) [inline app/src/main/java/com/uptech/example/vidme/ui/front/FollowingTabFragment.java]: use finallyDo

File: app/src/main/java/com/uptech/example/vidme/api/ApiService.java [hunk 3]
```diff
@@ -1,66 +1,91 @@
...
+    }
+
+    // add a handler that will create a NotLoggedInEvent
+    // if the user tries to perform an action that requires authorization
+    private <T> Observable <T> addOnErrorHandling(Observable <T> observable) {
+        return observable.doOnError(error -> {
+            if (error instanceof HttpException) {
+                HttpException exception = ((HttpException) error);
+                if (exception.code() == NetworkUtils.NOT_AUTHORIZED_CODE) {
+                    eventBus.post(new NotLoggedInEvent());
```
Comment (AndriyBas) [inline app/src/main/java/com/uptech/example/vidme/api/ApiService.java]: @HalynaHalkina @Dima564 let's consider adding some singleton that would handle this, smth like `NavigationHandler` or `ErrorHandler`

File: app/src/main/java/com/uptech/example/vidme/ui/front/NewTabFragment.java [hunk 4]
```diff
@@ -6,24 +6,18 @@
...
-                .subscribeOn(Schedulers.io())
-                .observeOn(AndroidSchedulers.mainThread())
-                .subscribe(videoListResponse -> {
-                    isLoading = false;
-                    adapter.addAll(videoListResponse.getVideos());
-                }, error -> {
-                    isLoading = false;
+                .doAfterTerminate(() -> isLoading = false)
+                .subscribe(videoListResponse ->
+                    adapter.addAll(videoListResponse.getVideos()), error -> {
```
Comment (AndriyBas) [inline app/src/main/java/com/uptech/example/vidme/ui/front/NewTabFragment.java]: @HalynaHalkina we need to check that fragment is attached

File: app/src/main/java/com/uptech/example/vidme/ui/login/LoginActivity.java [hunk 5]
```diff
@@ -108,71 +104,55 @@ private boolean checkUserInput(String userName, String password){
...
-                            break;
-                        default:
-                            makeToast(R.string.login_error);
-                    }
-                }
-                updateUi();
-            }
+        if (checkUserInput(userName, password)) {
+
+            apiService.authCreate("Base: " + encodeBase64(userName, password), userName, password)
```
Comment (AndriyBas) [inline app/src/main/java/com/uptech/example/vidme/ui/login/LoginActivity.java]: @HalynaHalkina @Dima564 this creation of `Base` header should not be here, encapsulate it somewhere

**Pull request merged**